frontend/src/reducers/property.js

```diff
@@ -55,6 +60,30 @@ const propertyState = (state = initialState, action) => {
           [action.fieldName]: action.newValue,
         },
       });
+    case "SET_ARRAY_VALUES":
+      let index = state.fields[action.arrayName].indexOf(action.item);
+      if (index > -1) {
+        return Object.assign({}, state, {
+          ...state,
+          fields: {
+            ...state.fields,
+            [action.arrayName]: state.fields[action.arrayName].filter(
+              (item) => item !== action.item
+            ),
+          },
+        });
+      } else {
+        return Object.assign({}, state, {
+          ...state,
+          fields: {
+            ...state.fields,
+            [action.arrayName]: [
+              ...state.fields[action.arrayName],
+              action.item,
+            ],
+          },
+        });
+      }
```

Comment:
we should have one reducer `SET_ADD_PROPERTY_FIELD`

this can have a condition:

```js
Array.isArray($field) {
  return ...state // merged in array
}
// else return normal
return {
  ...state
  ..
}
```